File: dataset_reduzido/avatar/Python/Code/codeforces_106_B.py
n = int ( input ( ) )
input_array = [ ]
for i in range ( n ) :
    S , R , H , C = map ( int , input ( ).split ( ) )
    input_array.append ( ( S , R , H , C ) )
outdated = [ ]
for i in range ( n ) :
    for j in range ( n ) :
        if ( input_array [ i ] [ 0 ] < input_array [ j ] [ 0 ] and input_array [ i ] [ 1 ] < input_array [ j ] [ 1 ] and input_array [ i ] [ 2 ] < input_array [ j ] [ 2 ] ) :
            outdated.append ( input_array [ i ] )
outdated = set ( outdated )
minimum_cost = 100000
input_number = 0
for i in range ( n ) :
    current_item = input_array [ i ]
    cost = current_item [ 3 ]
    if ( current_item not in outdated and cost < minimum_cost ) :
        minimum_cost = current_item [ 3 ]
        input_number = i + 1
print ( input_number )

File: output/Qwen_repaired_1/avatar/Python/Java/codeforces_106_B.java
import java.util.*;

public class codeforces_106_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        List<int[]> inputArray = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            String[] parts = scanner.nextLine().split(" ");
            int s = Integer.parseInt(parts[0]);
            int r = Integer.parseInt(parts[1]);
            int h = Integer.parseInt(parts[2]);
            int c = Integer.parseInt(parts[3]);
            inputArray.add(new int[]{s, r, h, c});
        }
        List<int[]> outdated = new ArrayList<>();
        for (int i = 0; i < inputArray.size(); i++) {
            for (int j = i + 1; j < inputArray.size(); j++) {
                int[] item1 = inputArray.get(i);
                int[] item2 = inputArray.get(j);
                if (item1[0] < item2[0] && item1[1] < item2[1] && item1[2] < item2[2]) {
                    outdated.add(item1);
                }
            }
        }
        outdated.retainAll(inputArray);
        Collections.reverse(outdated);
        Comparator<int[]> comparator = Comparator.comparingInt(arr -> arr[3]).reversed();
        outdated.sort(comparator);
        int minimumCost = 100000;
        int inputNumber = 0;
        for (int i = 0; i < inputArray.size(); i++) {
            int[] currentItem = inputArray.get(i);
            int cost = currentItem[3];
            if (!outdated.contains(currentItem) && cost < minimumCost) {
                minimumCost = cost;
                inputNumber = i + 1;
            }
        }
        System.out.println(inputNumber);
    }
}